{"timestamp": "2025-07-30T15:10:41.560961", "level": "ERROR", "category": "WEBUI", "component": "kassia.webui", "message": "WebSocket error", "details": {"error": "Cannot call \"send\" once a close message has been sent.", "client_ip": "127.0.0.1", "exception": {"type": "RuntimeError", "message": "Cannot call \"send\" once a close message has been sent.", "traceback": ["Traceback (most recent call last):\n", "  File \"C:\\Users\\z004ac7v\\Kassia\\Python\\web\\app.py\", line 779, in websocket_endpoint\n    await websocket.send_text(json.dumps(heartbeat))\n", "  File \"C:\\Users\\z004ac7v\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\starlette\\websockets.py\", line 166, in send_text\n    await self.send({\"type\": \"websocket.send\", \"text\": data})\n", "  File \"C:\\Users\\z004ac7v\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\starlette\\websockets.py\", line 98, in send\n    raise RuntimeError('Cannot call \"send\" once a close message has been sent.')\n", "RuntimeError: Cannot call \"send\" once a close message has been sent.\n"]}}}
{"timestamp": "2025-07-30T15:10:47.359408", "level": "ERROR", "category": "WEBUI", "component": "kassia.webui", "message": "WebSocket error", "details": {"error": "Cannot call \"send\" once a close message has been sent.", "client_ip": "127.0.0.1", "exception": {"type": "RuntimeError", "message": "Cannot call \"send\" once a close message has been sent.", "traceback": ["Traceback (most recent call last):\n", "  File \"C:\\Users\\z004ac7v\\Kassia\\Python\\web\\app.py\", line 779, in websocket_endpoint\n    await websocket.send_text(json.dumps(heartbeat))\n", "  File \"C:\\Users\\z004ac7v\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\starlette\\websockets.py\", line 166, in send_text\n    await self.send({\"type\": \"websocket.send\", \"text\": data})\n", "  File \"C:\\Users\\z004ac7v\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\starlette\\websockets.py\", line 98, in send\n    raise RuntimeError('Cannot call \"send\" once a close message has been sent.')\n", "RuntimeError: Cannot call \"send\" once a close message has been sent.\n"]}}}
